### DISCUSSION TITLE
Simple profiling mode for spikes in RAM use

### DISCUSSION THREAD

**Discussion by Epigene** [test-prof/test-prof#274] [Ideas]
Hi!

This may already be possible in test-prof, but the idea is to run the whole RSpec suite on CI with eager-load on and measure RAM use before and after each spec file (or even example). Would be nice to be able to specify the offending RAM use increase in value or percent. For example:

```
# report spec files that bumped RAM use by 100mb or more
TEST_MEM_PROF=100mb bundle exec rspec

# report spec files that bumped RAM use by 20% or more relative to starting objectspace size
TEST_MEM_PROF=20 bundle exec rspec
```

We're facing inexplicable RAM use spikes on CI and will roll a custom hook for now.

**Comment by palkan** [comment]
Hey!

Thanks for the suggestion. That would be a nice addition to TestProf. If you already have some script to share, it would help us to get it done faster.

**Comment by palkan** [comment]
Btw, I think, we can get some inspiration/functionality from https://github.com/gazay/mnemonic

**Comment by palkan** [comment]
Here you go: https://test-prof.evilmartians.io/profilers/memory_prof

(Will be released soon, already available in master)
